import "db_message.proto";

enum MSG_ACTION_TYPE
{
    MSG_REQUEST = 0;
    MSG_RESPONSE = 1;
    MSG_NOTIFY= 2;
}


enum MessageType
{
    //gate server start......
    MSG_SOCCER_PLAYER_INFO = 1;
    MSG_HEART_BEAT          = 2;//心跳
    MSG_CLIENT_LOGIN        = 3;//client login
    MSG_GATE_SERVER_REGISTER = 4;//gate server register
    MSG_INIT_CLIENT         = 5;
    MSG_CLIENT_MOVE     = 6;
}

enum DbMessageType
{
    MSG_DB_COMMON   = 10000;
}

message ClientHeartBeatRequest
{
    required int32 client_time = 1;
}

message ClientHeartBeatResponse
{
    required int32 server_time = 1;
}


message ClientLoginRequest
{
    required int32 role_id = 1;
    required string token = 2;
}

message ClientLoginResponse
{
    required int32 ret = 1;
}

message ClientInitNotf
{
    required  int32 ret = 1;
    required  DBUserInfo user_info = 2;
}

message ClientMoveRequest
{
    required int32 map_id = 1;
    required int32 pos_x = 2;
    required int32 pos_y = 3;
}

message ClientMoveResponse
{
    required int32 ret = 1;
    required int32 map_id = 2;
    required int32 pos_x = 3;
    required int32 pos_y = 4;
}
